### DISCUSSION TITLE
FindIntegrationAnchors does not converge

### DISCUSSION THREAD

**Discussion by HElias01** [stuart-lab/signac#1201] [Q&A] [answered]
Hi  @timoast, 

I attempted to integrate a Multiome dataset (generated using your 10xmultime vignette) with scATACseq using the scATAC-seq data integration vignette, unfortunately, I am running into issues. 

**_This is the error:_** 

### FindingIntegrationAnchors

Warning: Some cell names are duplicated across objects provided. Renaming to enforce unique cell names. Computing within dataset neighborhoods
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=02s  
Finding all pairwise anchors
  |                                                  | 0 % ~calculating  Warning: No filtering performed if passing to data rather than countsProjecting new data onto SVD
Projecting new data onto SVD
Finding neighborhoods
Finding anchors
	Found 1136 anchors
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100% elapsed=30s 

### Running IntegrateEmbedding
Error: The cell names in the reduction provided don't match the cell names present in the objects used to build the AnchorSet

**_This is my code:_**
library(Signac)
library(Seurat)
library(ggplot2)

young1.multi <- readRDS("young1_multiome.rds")
fragpath2<- "/Users/eliash/Desktop/multiomics/HA-1570_Young_mice_2_11_atac_fragments.tsv.gz"
fragcounts <- CountFragments(fragments = fragpath2)
young2.cells <- fragcounts[fragcounts$frequency_count > 2000, "CB"]
young2atac.frags <- CreateFragmentObject(path = fragpath2, cells = young2.cells)

young2.counts <- FeatureMatrix(
  fragments = young2atac.frags,
  features = granges(young1.multi),
  cells = young2.cells)

young2atac.assay <- CreateChromatinAssay(
  counts = young2.counts,
  min.features = 1000,
  fragments = young2atac.frags)

young2.atac <- CreateSeuratObject(counts = young2atac.assay, assay = "peaks")
young2.atac <- subset(young2.atac, nCount_peaks > 2000 & nCount_peaks < 30000)

young2.atac <- FindTopFeatures(young2.atac, min.cutoff = 10)
young2.atac <- RunTFIDF(young2.atac)
young2.atac <- RunSVD(young2.atac)

young2.atac$dataset <- "ATAC"
young1.multi$dataset <- "multiome"
young.combined <- merge(young2.atac, young1.multi)

young.combined <- FindTopFeatures(young.combined, min.cutoff = 10)
young.combined <- RunTFIDF(young.combined)
young.combined <- RunSVD(young.combined)
young.combined <- RunUMAP(young.combined, reduction = "lsi", dims = 2:30)
p1 <- DimPlot(young.combined, group.by = "dataset")

integration.anchors <- FindIntegrationAnchors(
  object.list = list(young1.multi, young2.atac),
  anchor.features = rownames(young1.multi),
  reduction = "rlsi",
  dims = 2:30)

integrated <- IntegrateEmbeddings(
anchorset = integration.anchors,
reductions = young.combined[["lsi"]],
new.reduction.name = "integrated_lsi",
dims.to.integrate = 1:30)

Any suggestions?

**Comment by timoast** [comment, marked as answer]
If there are cell name conflicts between the objects you're integrating this can cause issues (https://github.com/timoast/signac/issues/923)

The workaround is to rename cells in each object so that there are no duplicate cell names across the objects, before performing integration. eg:
1. Rename cells in each object to add a unique prefix so that there are no name collisions across objects to be merged
2. Merge the objects and create the un-corrected LSI dimension reduction
3. Find integration anchors
4. Use the integration anchors to correct the LSI embeddings
